### PR TITLE
WebKit-based web view controller

### LINKED ISSUE #7781: Migrate WPWebViewController to WKWebView
The logic on `WPWebViewController` is not very clear, and we would need to add some fixes to it to support https://github.com/wordpress-mobile/WordPress-iOS/issues/7759 and https://github.com/wordpress-mobile/WordPress-iOS/issues/7761, as specified in https://github.com/wordpress-mobile/WordPress-iOS/issues/7778
However, considering https://github.com/wordpress-mobile/WordPress-iOS/issues/6371 and the future, it seems more reasonable to re-write it as a WKWebView in Swift (careful: due to the bug mentioned on https://github.com/wordpress-mobile/WordPress-iOS/issues/6371 we may only be able to use it on clients on iOS 11 and over)

cc: @koke

### PR DESCRIPTION
**Pull request by koke**
Implements a new `WebKitViewController` that will be used on devices running iOS 11 and above. We're keeping the old `WPWebViewController` for iOS 10 devices.

The sharing flow is still using `WPWebViewController` since it uses some custom logic, and it might be more interesting to switch to SFAuthenticationSession (#7854) than porting the custom logic to the new navigation delegate methods.

The design of this might not be the most elegant, but there were quite a few constraints here:
- iOS 10 and iOS 11
- UIWebView and WebKit
- Objective-C and Swift

After some experiments with a protocol based approach that could not work because of the mix of languages, the simplest remaining approach was to use a factory class, and provide every possible input wrapped in a configuration object. The factory then returns a UIViewController that's enough, since the callers don't need any further customization after the initial configuration.

## For design review

Since I had to recreate the whole view controller, I took the liberty to change the UI so it's closer to the native Safari controller:

- The toolbar is always visible if configured (previous version only showed it after navigation)
- Moved the share button to the toolbar
- Added "reload" button
- Added a "globe" icon to open in Safari. If we want the actual Safari icon here, we'll have to bundle one.

Here's a comparison of old, new, and Safari:

webkitvc-safari

Fixes #7781, #6371, #4381

## To test

Test in general that everything using a web VC still works as expected. If you don't have an iOS 10 device/simulator, you can tweak the `if` in `WebViewControllerFactory` so it always returns a `WPWebViewController`.

Needs review: @elibud

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Stats/JetpackLoginViewController.swift [hunk 1]
```diff
@@ -451,12 +451,8 @@ class JetpackLoginViewController: UIViewController {
...
             return
         }
-        guard let webViewController = WPWebViewController(url: url) else {
-            return
-        }
-
-        webViewController.authenticate(with: blog)
-
+        let webViewController = WebViewControllerFactory.controller(url: url, blog: blog)
+
```
[the remark this answers is not in the document]
Comment (elibud) [reply WordPress/Classes/ViewRelated/Stats/JetpackLoginViewController.swift]: 😱

File: WordPress/Classes/Utility/WebKitViewController.swift [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+        url = configuration.url
+        customOptionsButton = configuration.optionsButton
+        secureInteraction = configuration.secureInteraction
+        addsWPComReferrer = configuration.addsWPComReferrer
+        customTitle = configuration.customTitle
+        authenticator = configuration.authenticator
+        navigationDelegate = configuration.navigationDelegate
+
+        super.init(nibName: nil, bundle: nil)
+        hidesBottomBarWhenPushed = true
```
[the remark this answers is not in the document]
Comment (koke) [reply WordPress/Classes/Utility/WebKitViewController.swift]: Sort of, I think. You have to initialize all properties defined in the subclass before calling `init`, and it's not a good idea (not sure if the compiler enforces it) to touch any of the parent class properties before calling `super.init`. I think part of the confusion is that some properties (secureInteraction, addsWPComReferrer) already had an initial value, so I'll change those to `let`.

File: WordPress/Classes/Utility/WebKitViewController.swift [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+        }
+        webView.load(request)
+    }
+
+    func configureNavigation() {
+        let closeButton = UIBarButtonItem(image: Gridicon.iconOfType(.cross), style: .plain, target: self, action: #selector(WebKitViewController.close))
+        closeButton.accessibilityLabel = NSLocalizedString("Dismiss", comment: "Dismiss a view. Verb")
+
+        // Only show a close button if this is presented modally
+        if (navigationController?.viewControllers.count == 1 && presentingViewController != nil) {
```
[the remark this answers is not in the document]
Comment (koke) [reply WordPress/Classes/Utility/WebKitViewController.swift]: Making it a method in a `UIViewController` can definitely be more descriptive, but having two instances of it is a mistake 😬 I added this one with my latest changes and forgot I was already doing that check below.

**Comment by koke** [comment]
> looking at the code it looks like you did this on purpose, is that right? I don't oppose it but I'm not sure there's a big gain in doing it this way either.

Yes it was on purpose. I agree it looks a bit odd, maybe replacing the reload icon is best in that case. I'll defer to the design review for that though.

**Comment by mattmiklic** [comment]
Design looks good to me. 

On the Safari icon, agreed that the globe is less clear, but I think including a Safari icon could run into trademark issues with Apple. It could also look a bit out of place visually, since our icons are drawn with a 2pt stroke, and Apple's are drawn with a 1px stroke. Given that, I think the globe icon works.

**Pull request merged**